frontend/src/metabase/collections/containers/CollectionSidebar.jsx

```diff
@@ -41,16 +42,17 @@ const Sidebar = styled(Box)`
     we should eventually refactor code elsewhere in the app to use this by default instead of determining the relationships clientside, but this works in the interim
   */
   query: () => ({ tree: true }),
+  loadingAndErrorWrapper: false,
 })
 class CollectionSidebar extends React.Component {
   state = {
     openCollections: [],
   };
 
-  componentDidMount() {
-    // an array to store the ancestors
+  componentDidUpdate(prevProps) {
     const { collectionId, collections, loading } = this.props;
-    if (!loading) {
+    const loaded = prevProps.loading && !loading;
+    if (loaded) {
       const ancestors = getParentPath(collections, Number(collectionId)) || [];
       this.setState({ openCollections: ancestors });
     }
```

Comment:
Replacing `didMount` with `didUpdate` ended up in one of the tests failing
Going to look at it tomorrow, until then putting the PR into draft

Reply:
Funny thing, I guess it just made it better. So `didMount` made sure the correct tree "branch" is expanded when we go to a specific nested collection.

The reason `didMount` is replaced `didUpdate` is that entity loader's `loadingAndErrorWrapper` is now disabled for `CollectionSidebar`. In that way, we can display the loader that won't cause the UI jump. But having `loadingAndErrorWrapper` enabled guaranteed that the sidebar is rendered only after its data is loaded, so the `loading` prop was always `false`. Now when the sidebar is responsible for loading state handling, it mounts in the `loading` state, and the method body won't be executed.

So now ancestors are recalculated each time sidebar's data changes. It worked well with this logic in the `didMount`, but the failing test catched the edge case

So imagine I have a collection structure like this:

<img width="336" alt="initial-state" src="https://user-images.githubusercontent.com/17258145/115761156-b4eb0100-a3aa-11eb-9e70-428939071864.png">

I want to move "Second collection" to the "New collection". So I select the "Second collection", click edit, and change its parent collection. On the `release-x.39.x` I sometimes appear in the next state after I move the collection:

<img width="333" alt="release-after-move" src="https://user-images.githubusercontent.com/17258145/115761336-eb288080-a3aa-11eb-8d21-3ae652a58b64.png">

But on this branch, after the same steps I appear in the next state:

<img width="334" alt="fix-after-move" src="https://user-images.githubusercontent.com/17258145/115761410-faa7c980-a3aa-11eb-9665-b52d671bfd49.png">

So I guess it's actually better